[18.0][MIG] sale_order_amount_to_invoice: Migration to 18.0

Anushapadinharepurakkal opened the pull request.

leemannd committed:
base module & sales_team

Add base for wizard

wizard for manual procurements

Update base computation

update to OCA guidelines

jcoux committed:
Add action_manual_procurement_wizard

Add action_manual_procurement_wizard in view

leemannd committed:
flake8 & OCA standards

jgrandguillaume committed:
[IMP] report manual procurement on SO from sale team

[ADD] Translation and .pot file

jgrandguillaume committed:
[TYP] Change manual proc for manual delivery

jgrandguillaume committed:
[IMP] Toggle manual,standard as a button in SO + UI

[IMP] Update translation files

[IMP] lot's of functionnal improvments

Don't add already planned or shipped lines in wizard
Ensure you have one picking per manual delivery
Forbid to add lines or ship more than ordered
Planned date is required
Improve views, naming and buttons
Better Readme
Default Carrier set from SO

[FIX] Do not add product of type 'service' in manual delivery creation

jgrandguillaume committed:
[ADD] Tests on various use cases

mdietrichc2c committed:
Regroup picking per date_planned / carrier + correct wizard action from SO

Add product name to manual delivery lines

vrenaville committed:
[FIX] picking split criteria

[FIX] call super in case of standard delivery

[FIX] when multiple picking candidate available take the right one with date corresponding

[FIX] revert of renaming field

Add SO line description

Commit:
add Delivery Address to wizard. by default selec SO partner address.

rousseldenis committed:
[10.0][ADD] sale_invoice_policy

rousseldenis committed:
[IMP] Add search method, secures _get_to_invoice_qty,compute_invoice_status computation

rousseldenis committed:
[IMP] sale_invoice_policy: Implement inverse

rousseldenis committed:
[IMP] sale_invoice_policy: Let the availability of choosing no policy on sale order

rousseldenis committed:
[IMP] sale_invoice_policy: Add config for default_policy and if policy is required on Sale Orders

sergio-teruel committed:
[11.0][ADD] sale_elaboration: New module

sergio-teruel committed:
[11.0][ADD] sale_elaboration: Display elaboration code in picking

sergio-teruel committed:
[11.0][ADD] sale_elaboration: Display salesman in line views

acsonefho committed:
[FIX] Fix perf issue during compute invoice status (of sale.order.line)

sergio-teruel committed:
[11.0][FIX] sale_elaboration: Get first elaboration product

sergio-teruel committed:
[11.0][IMP] sale_elaboration: Add to invoice line the sale order reference

pedrobaeza committed:
[IMP] sale_elaboration: Some improvements

sergio-teruel committed:
[11.0][ADD] sale_elaboration: Add is_elaboration in product template

pedrobaeza committed:
[IMP] sale_elaboration: Ensure test correctness + hide is_elaboration if not service

jjscarafia committed:
[11.0][FIX] sale_elaboration should't be auto installed

marylla committed:
Added translation using Weblate (German)

sergio-teruel committed:
[IMP] sale_elaboration: Reset "is_elaboration" in order line when user changes product

carlosdauden committed:
[IMP] sale_elaboration: Set is_elaboration in product tests

Tonow-c2c committed:
[MIG][11] rm coding* + python 2to3

[MIG][11] Update Readme + Manifest

[MIG][11] Adapt procurement for V11 + rm api.one

Tonow-c2c committed:
[MIG][11] Run black to fix lint

Tonow-c2c committed:
[MIG][11] Fix tests for V11 stock.picking

vrenaville committed:
[FIX] can switch from manual on sent state + fix field name changed

oca-travis committed:
[UPD] Update sale_invoice_policy.pot

OCA-git-bot committed:
[ADD] icon.png

sergio-teruel committed:
[FIX] sale_elaboration: Apply force_save for field is_elaboration in sale order view to store the value in onchange method

grindtildeath committed:
Remove unneeded calls to invalidate_cache

Fix view name and improve tests

IMP: Use a procurement group according to date planned

Fix manifest keys

AaronHForgeFlow committed:
[IMP]allow to manually deliver sale order lines from different sale orders

AaronHForgeFlow committed:
[MIG]sale_manual_delivery to v12

[IMP]sale_manual_delivery. Specify route at the time of the delivery

sergio-teruel committed:
[MIG] sale_elaboration: Migration to v12.0

AaronHForgeFlow committed:
[MIG]sale_invoice_policy to v12

oca-travis committed:
[UPD] Update sale_invoice_policy.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/

sergio-teruel committed:
[IMP] sale_elaboration: black, isort

sergio-teruel committed:
[MIG] sale_elaboration: Migration to v13.0

mmequignon committed:
[FIX] Restore res_partner field in the view with respect to last changes.

Tonow-c2c committed:
[FIX] Partner id will be retrive from wizard to procurement rule

Tonow-c2c committed:
[FIX] wizard float_compare product precision

Tonow-c2c committed:
[FIX] to use the right product uom rounding

georginaf committed:
Added translation using Weblate (Catalan)

georginaf committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/ca/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/

pedrocs-exo committed:
Added translation using Weblate (Portuguese)

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/pt/

renatonlima committed:
[FIX] pep8 and pylint

SandieFavre committed:
Translated using Weblate (French)

Currently translated at 70.6% (12 of 17 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/fr/

quentinDupont committed:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_invoice_policy/fr/

oca-travis committed:
[UPD] Update sale_manual_delivery.pot

OCA-git-bot committed:
sale_manual_delivery 12.0.2.0.1

danimv5 committed:
Added translation using Weblate (Spanish)

danimv5 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (48 of 48 strings)

Translation: sale-workflow-12.0/sale-workflow-12.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-12-0/sale-workflow-12-0-sale_manual_delivery/es/

alexeirivera87 committed:
[IMP] sale_invoice_policy: black, isort, prettier

alexeirivera87 committed:
[MIG] sale_invoice_policy: Migration to 13.0

ivantodorovich committed:
[MIG] sale_manual_delivery: Migration to 13.0

ivantodorovich committed:
[IMP] sale_manual_delivery: black, isort, prettier

oca-travis committed:
[UPD] Update sale_manual_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update sale_invoice_policy.pot

OCA-git-bot committed:
sale_invoice_policy 13.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

vrenaville committed:
[FIX] for drop shipping module

santostelmo committed:
[13.0] Fix view_sales_order_undelivered_line_filter

OCA-git-bot committed:
sale_manual_delivery 13.0.1.0.1

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 100.0% (17 of 17 strings)

Translation: sale-workflow-13.0/sale-workflow-13.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-13-0/sale-workflow-13-0-sale_invoice_policy/es_MX/

OCA-git-bot committed:
sale_manual_delivery 13.0.1.0.2

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_quotation_number
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_quotation_number/de/

nilshamerlinck committed:
[FIX] sale_isolated_quotation and sale_quotation_number are incompatible: they should not be installed nor tested together

JordiMForgeFlow committed:
[13.0][ADD] sale_order_line_delivery_state

oca-travis committed:
[UPD] Update sale_order_line_delivery_state.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_quotation_number 14.0.1.0.3

hailangvn committed:
[IMP] sale_quotation_number: refactor for sale_isolated_quotation

This is to resolve conflict with `sale_isolated_quotation`. Then that
module will depends on this module.

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_quotation_number 14.0.2.0.0

bizzappdev committed:
[IMP] sale_quotation_number : black, isort, prettier

bizzappdev committed:
[MIG]sale_quotation_number : Migrate sale_quotation_number to v15.

ChrisOForgeFlow committed:
[IMP] sale_manual_delivery: black, isort, prettier

ChrisOForgeFlow committed:
[14.0][MIG] sale_manual_delivery

ChrisOForgeFlow committed:
[ADD] init hook script for big databases

ChrisOForgeFlow committed:
[ADD] permissions on trasient models

CarlosRoca13 committed:
[IMP] sale_elaboration: Add menu entry to sale elaboration report at inventory reporting

ChrisOForgeFlow committed:
[IMP] added field to don't show button if not exists pending to deliver

yankinmax committed:
[IMP] sale_invoice_policy: black, isort, prettier

yankinmax committed:
[15.0][MIG] sale_invoice_policy: Migration to 15.0

yankinmax committed:
[FIX] sale_invoice_policy: linter issues

Commit:
[UPD] Update sale_invoice_policy.pot

Commit:
[UPD] Update sale_quotation_number.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
sale_quotation_number 15.0.1.0.1

ChrisOForgeFlow committed:
[FIX] user validation error, update definition act_window

DhavalTforgeFlow committed:
[MIG] sale_order_line_delivery_state: Migration to 15.0

CarlosRoca13 committed:
[MIG] sale_elaboration: Migration to 14.0

Commit:
[UPD] Update sale_order_line_delivery_state.pot

OCA-git-bot committed:
sale_order_line_delivery_state 15.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[IMP] sale_elaboration: Improve elaboration tree view

sergio-teruel committed:
[FIX] sale_elaboration: Fix delivery slip report. Add config setting to allow users display elaboration notes on delivery slips

CarlosRoca13 committed:
[MIG] sale_elaboration: Migration to v15

sergio-teruel committed:
[IMP] sale_elaboration: Add pre-init hooks to quickly install

Commit:
[IMP] sale_elaboration: Allow to select multiple elaborations

sergio-teruel committed:
[IMP] sale_elaboration: Improve form view and kanban view on sale order lines

sergio-teruel committed:
[IMP] sale_elaboration: Allow to determine a stock.location.route for the elaborations

pedrobaeza committed:
[IMP] sale_invoice_policy: Don't repeat selection values

WARNING odoo odoo.fields: res.config.settings.sale_default_invoice_policy: selection attribute will be ignored as the field is related

OCA-git-bot committed:
sale_invoice_policy 15.0.1.0.1

sergio-teruel committed:
[IMP] sale_elaboration: Improve kanban views

ljsalvatierra-factorlibre committed:
[MIG] sale_invoice_policy: Migration to 16.0

Commit:
[MIG] sale_manual_delivery: Migration to 16.0

Commit:
[IMP] sale_manual_delivery: test checking has_pending_delivery

Commit:
[RM] sale_manual_delivery: remove checking state of lines

sergio-teruel committed:
[IMP] sale_elaboration: Add elaboration mixin and add feature in pickings

sergio-teruel committed:
[IMP] sale_elaboration: Add elaboration profiles

sergio-teruel committed:
[IMP] sale_elaboration: Allow multi edit in elaborations

[FIX] sale_elaboration: name_search don't search by name after change rec_name to code

sergio-teruel committed:
[IMP] sale_elaboration: Compute sale elaboration price in sale order line

Commit:
[UPD] Update sale_invoice_policy.pot

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[FIX] sale_elaboration: Merge moves with distinct elaboration
TT43660

Commit:
[MIG] sale_elaboration: Migration to 16.0

sergio-teruel committed:
[FIX] sale_invoice_policy: Wrong inheritance field with precompute attribute

dsolanki-initos committed:
[MIG] sale_quotation_number: Migration to 16.0

simoneversienti committed:
[fix] sale_quotation_number: fix multi-company

Commit:
[UPD] Update sale_manual_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_manual_delivery/

Commit:
[UPD] Update sale_quotation_number.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_quotation_number
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_quotation_number/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (49 of 49 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (48 of 48 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_manual_delivery/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Shide committed:
[ADD] sale_invoice_frequency

OCA-git-bot committed:
[UPD] README.rst

Shide committed:
[FIX] sale_invoice_frequency: name translatable

OCA-git-bot committed:
[UPD] README.rst

oihane committed:
[MIG] sale_order_line_delivery_state: Migration to 16.0

Commit:
[UPD] Update sale_order_line_delivery_state.pot

dhp-denero committed:
[fix] sale_manual_delivery: wrong route_id model

``stock.location.route`` is no more available.

It is replaced with stock.route

Exception Occurs when trying to Popup Create Manual Delivery Due to route_id relational model is not available.

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_invoice_policy/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_line_delivery_state
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_line_delivery_state/es/

Commit:
[16.0][FIX] sale_invoice_policy: compute qty to invoice

rousseldenis committed:
[UPD] sale_invoice_policy: Don't install it anymore

As the invoice_policy is computed non stored, it makes errors occurs
in transversal modules

CarlosRoca13 committed:
[IMP] sale_elaboration: Add field to filter elaborated lines

Shide committed:
[FIX] sale_invoice_frequency: Add Monthly

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_quotation_number
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_quotation_number/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

Shide committed:
[FIX] sale_invoice_frequency: Add `invoice_frequency_id` to commercial fields

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (10 of 10 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_line_delivery_state
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_line_delivery_state/it/

yajo committed:
[IMP] sale_elaboration: optional subtree fields

These views have more than enough fields. It's good to be able to hide some.

@moduon MT-3011

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_invoice_policy/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (48 of 48 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_manual_delivery/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (49 of 49 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/it/

bosd committed:
[IMP] sale_quotation_number: deprecated call to Savepointcase

OCA-git-bot committed:
[BOT] post-merge updates

dessanhemrayev committed:
[ADD] sale_order_amount_to_invoice: Add module

[IMP] sale_order_amount_to_invoice: updated logic for calculating the amount

[UPD] Update sale_order_amount_to_invoice.pot

[BOT] post-merge updates

Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_order_amount_to_invoice
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_order_amount_to_invoice/it/

OCA-git-bot committed:
[BOT] post-merge updates

astirpe committed:
[17.0][MIG] sale_manual_delivery: pre-commit

astirpe committed:
[17.0][MIG] sale_manual_delivery: Migration to 17.0

Shide committed:
[FIX] sale_elaboration: Do not print Elaboration on incoming pickings

yajo committed:
[FIX] sale_elaboration: propagate note-only lines to stock moves

Before this patch, if a sale order line had no elaborations selected, but had elaboration notes, those notes didn't land in the created `stock.move` records when confirming the SO.

@moduon MT-5511

yajo committed:
[IMP] sale_elaboration: remove elaboration column from delivery slip

Elaboration notes were not displayed in the delivery slip, but adding them in the preexisting column could make the column too big.

I removed the column and, instead, print the elaboration notes after the product name, where there's enough room for more text. As a side effect, the delivery slip is now more compact and easier to read.

@moduon MT-5508

yajo committed:
[IMP] sale_elaboration: display elaboration notes in picking operations report

@moduon MT-5508

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (51 of 51 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/it/

EmilioPascual committed:
[IMP] sale_elaboration: Add elaborations and elaboration note in move line tree detailed

@moduon MT-5810

EmilioPascual committed:
[IMP] sale_elaboration: Add icon fa-comment to reports

MT-5814 @moduon

Gelojr committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (50 of 50 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/es/

Commit:
[UPD] Update sale_invoice_policy.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_invoice_policy
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_invoice_policy/

Commit:
[UPD] Update sale_manual_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-17.0/sale-workflow-17.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-17-0/sale-workflow-17-0-sale_manual_delivery/

yajo committed:
[FIX] sale_elaboration: use parent.state in domain

Without this patch, and with some unknown but feasible combination of sale-workflow modules, these steps were producing an exception:

1. Switch to mobile view.
2. Edit one sale order line.
3. Change packaging.
4. Save line (but not parent record).
5. Repeat steps 2-4.

The error was:

    modifier "readonly": Unknown field state in domain

I'm not sure why just `state` wasn't working reliably, but I can tell Odoo upstream uses `parent.state` since https://github.com/odoo/odoo/commit/a7883df9eac4bd93417846a44f185e6c2bc36eca and this fixes the issue.

@moduon MT-5351

yajo committed:
[FIX] sale_elaboration: compatible with delivery (don't compute notes)

Under certain circumstances, elaboration notes were deleted. For example, when adding or removing an elaboration.

That's more or less OK, but also when delivery is installed along with this module, then the notes removal is unpredictable.

Here I fix https://github.com/OCA/sale-workflow/issues/3154 by removing the computation on notes.

Until now, notes were just repeating the same as we had in elaborations. Since the information was duplicated, the `elaboration_ids` field never appeared in reports.

Now the notes are just a normal text field that does nothing if empty. Since it will be usually empty, now reports need to include `elaboration_ids` too.

@moduon MT-6164

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (50 of 50 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (47 of 47 strings)

Translation: sale-workflow-17.0/sale-workflow-17.0-sale_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-17-0/sale-workflow-17-0-sale_manual_delivery/it/

Shide committed:
[FIX] sale_elaboration: Fix report call

pilarvargas-tecnativa committed:
[IMP] sale_elaboration: Add delivery date to the elaborations report

TT49619

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (54 of 54 strings)

Translation: sale-workflow-16.0/sale-workflow-16.0-sale_elaboration
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-16-0/sale-workflow-16-0-sale_elaboration/it/

Commit:
[IMP] sale_quotation_number: pre-commit auto fixes

Commit:
[MIG] sale_quotation_number: Migration to 17.0

Commit:
[UPD] Update sale_quotation_number.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-17.0/sale-workflow-17.0-sale_quotation_number
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-17-0/sale-workflow-17-0-sale_quotation_number/

Commit:
[FIX] sale_quotation_number: fix revision naming for sale orders

CRogos committed:
[IMP] sale_quotation_number: allow customizable quotation sequence

rousseldenis committed:
[FIX] sale_invoice_policy: Don't use the 'required' parameter

Don't use the 'required' parameter to compute the quantities to invoice or not.
In fact, this should only depends on the invoice policy itself.

rousseldenis committed:
[IMP] sale_invoice_policy: Don't reuse fields in depends

As ORM appends new depends() parameters to existing ones, simplify the writing

rousseldenis committed:
[IMP] sale_invoice_policy: Don't rely on required option to compute the amount

As for the 'qty_to_invoice' field, the 'untaxed_amount_to_invoice' field should not
depends on the required option to compute its value

rousseldenis committed:
[IMP] sale_invoice_policy: Don't rewrite Odoo core compute method

As we don't want to rewrite Odoo core method inside this module, we
should not rewrite the method to compute the untaxed amount to invoice.

Instead, save the product invoice policy, change it to sale order one without
recomputing context, call the super() method and then, restore its former value.

rousseldenis committed:
[IMP] sale_invoice_policy: save products invoice policy values in a separate dict

rousseldenis committed:
[IMP] sale_invoice_policy: Add the default behavior as sale invoice policy

With default behavior in Odoo, the invoicing policy depends on products configuration.
So, include this as possible value for sale order invoicing policy. Possible values are so:

  - Products Invoice Policy: each product applies its invoicing policy (by sale line)
  - Ordered quantities: the whole sale order will apply the invoicing policy on ordered quantities
  - Delivered Quantities: the whole sale order will apply the invoicing policy on delivered quantities

That said, the 'required' configuration is no more useful as the invoicing policy is now required
on sale order level.

rousseldenis committed:
[IMP] sale_invoice_policy: Use protecting() instead of deprecated norecompute()

To avoid marking 'invoice_policy' product field as dirty for products temporary value,
use the working context manager 'protecting()' instead of deprecated 'norecompute() one.

Put the whole process in its own contextmanager.

rousseldenis committed:
[IMP] sale_invoice_policy: Improve README

OCA-git-bot committed:
[BOT] post-merge updates

marielejeune committed:
[IMP] sale_invoice_policy: pre-commit auto fixes

marielejeune committed:
[MIG] sale_invoice_policy: Migration to 18.0

OCA-git-bot committed:
[BOT] post-merge updates

Shide committed:
[FIX] sale_elaboration: Apply groups to views correctly

bizzappdev committed:
[MIG] sale_quotation_number: Migration to 18.0

bizzappdev committed:
[MIG] sale_manual_delivery: Migration to 18.0

Shide committed:
[FIX] sale_elaboration: Ensure elaboration line is created always in the Sale directly related to the move

ArnauCForgeFlow committed:
[IMP] sale_order_line_delivery_state: pre-commit auto fixes

ArnauCForgeFlow committed:
[MIG] sale_order_line_delivery_state: Migration to 18.0

CarlosRoca13 committed:
[IMP] sale_elaboration: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] sale_elaboration: Migration to 18.0

Commit:
[IMP] sale_order_amount_to_invoice: pre-commit auto fixes

Commit:
[MIG] sale_order_amount_to_invoice: Migration to 18.0

[IMP]sale_amount_to_invoice:missing files added

rousseldenis commented:
/ocabot migration sale_order_amount_to_invoice

ByteMeAsap reviewed (approved):
LGTM!!

ivs-cetmix reviewed (changes requested):
Hi @Anushapadinharepurakkal , thank you for your contribution.

Please ensure that all the steps of the [migration checklist](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0) are completed.
You should also squash administrative commits (if any) with the previous commit for reducing commit noise

Anushapadinharepurakkal commented:
@ivs-cetmix 

Can you please review the changes now

ivs-cetmix commented:
@Anushapadinharepurakkal this one is still there
![image](https://github.com/user-attachments/assets/fac0b9d3-6845-4ee9-aa23-121638cb7716)

JordiMForgeFlow committed:
[FIX] sale_stock_picking_blocking: copy method returns multiple records

rf-ife committed:
[18.0][FIX] sale_stock_picking_blocking: Fixed issue when duplicating a sales order.

rousseldenis commented:
@ivs-cetmix

Commit:
[UPD] Update sale_quotation_number.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

ivs-cetmix reviewed (approved):
Looks good now.

/ocabot merge nobump

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 11.9% (5 of 42 strings)

Translation: sale-workflow-18.0/sale-workflow-18.0-sale_exception
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-18-0/sale-workflow-18-0-sale_exception/nl/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (42 of 42 strings)

Translation: sale-workflow-18.0/sale-workflow-18.0-sale_exception
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-18-0/sale-workflow-18-0-sale_exception/nl/

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-workflow-18.0/sale-workflow-18.0-sale_force_invoiced
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-18-0/sale-workflow-18-0-sale_force_invoiced/tr/

Shide committed:
[IMP] sale_invoice_frequency: pre-commit auto fixes

Shide committed:
[MIG] sale_invoice_frequency: Migration to 18.0

rousseldenis commented:
@Anushapadinharepurakkal It seems you deleted `readme` fragments. Could you restore them ?

Commit:
[UPD] Update sale_invoice_frequency.pot

OCA-git-bot committed:
[BOT] post-merge updates

dreispt commented on sale_order_amount_to_invoice/__manifest__.py:
```diff
@@ -1,9 +1,8 @@
-# Copyright (C) 2024 Cetmix OÜ
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Don't remove copyright notices please.

dreispt commented on sale_order_amount_to_invoice/i18n/it.po:
```diff
@@ -1,34 +0,0 @@
-#. module: sale_order_amount_to_invoice
-#: model:ir.model.fields,field_description:sale_order_amount_to_invoice.field_sale_order__untaxed_amount_to_invoice
-msgid "Untaxed Amount To Invoice"
-msgstr "Importo impobibile da fatturare"
```
Don't remove translations.

OCA-git-bot committed:
[BOT] post-merge updates

dreispt reviewed (changes requested):
Copyright notices, translations and documentation should not be removed.

Commit:
[UPD] Update sale_order_line_delivery_state.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update sale_manual_delivery.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update sale_invoice_policy.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[UPD] addons table in README.md

Commit:
[UPD] Update sale_elaboration.pot

Commit:
[UPD] Update sale_order_line_menu.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-18.0/sale-workflow-18.0-sale_order_line_menu
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-18-0/sale-workflow-18-0-sale_order_line_menu/

Anushapadinharepurakkal commented:
This PR is currently failing due to the test script in the sale_order_product_availability_inline module

Anushapadinharepurakkal committed:
Merge branch '18.0' into 18.0-mig-sale_order_amount_to_invoice

Anushapadinharepurakkal commented:
Closing the Merge Request.Created a new One https://github.com/OCA/sale-workflow/pull/3862/
